cluster-autoscaler/utils/kubernetes/listers.go

```diff
@@ -144,30 +144,92 @@ type PodLister interface {
 	List() ([]*apiv1.Pod, error)
 }
 
+// isScheduled checks whether a pod is scheduled on a node or not
+func isScheduled(pod *apiv1.Pod) bool {
+	if pod == nil {
+		return false
+	}
+	return pod.Spec.NodeName != ""
+}
+
+// isDeleted checks whether a pod is deleted not
+func isDeleted(pod *apiv1.Pod) bool {
+	if pod == nil {
+		return false
+	}
+	return pod.GetDeletionTimestamp() != nil
+}
+
+// isUnschedulable checks whether a pod is unschedulable or not
+func isUnschedulable(pod *apiv1.Pod) bool {
+	if pod == nil {
+		return false
+	}
+	if isScheduled(pod) || isDeleted(pod) {
+		return false
+	}
+	_, condition := podv1.GetPodCondition(&pod.Status, apiv1.PodScheduled)
+	if condition == nil || condition.Status != apiv1.ConditionFalse || condition.Reason != apiv1.PodReasonUnschedulable {
+		return false
+	}
+	return true
+}
+
+// getIsDefaultSchedulerIgnored checks if the default scheduler should be ignored or not
+func getIsDefaultSchedulerIgnored(ignoredSchedulers map[string]bool) bool {
```

Comment:
Instead of checking this for every pod, maybe just extend the map once at startup? `if ignoredSchedulers[apiv1.DefaultSchedulerName] { ignoredSchedulers[""] = true }`.

Reply:
Updated `GetIgnoredSchedulersMap` to set that field